src/pgp.rs

```diff
@@ -99,13 +164,27 @@ pub fn create_keypair(addr: impl AsRef<str>) -> Option<(Key, Key)> {
     Some((Key::Public(public_key), Key::Secret(private_key)))
 }
 
-/// Select subkey of the public key to use for encryption.
+/// Select public key or subkey to use for encryption.
+///
+/// First, tries to use subkeys. If none of the subkeys are suitable
+/// for encryption, tries to use primary key. Returns `None` if the public
+/// key cannot be used for encryption.
 ///
-/// Currently the first subkey is selected.
-fn select_pk_for_encryption(key: &SignedPublicKey) -> Option<&SignedPublicSubKey> {
-    key.public_subkeys.iter().find(|_k|
-        // TODO: check if it is an encryption subkey
-        true)
+/// TODO: take key flags and expiration dates into account
+fn select_pk_for_encryption(key: &SignedPublicKey) -> Option<SignedPublicKeyOrSubkey> {
+    key.public_subkeys
+        .iter()
+        .find(|k| k.is_encryption_key())
+        .map_or_else(
+            || {
+                if key.is_encryption_key() {
```

Comment:
this check seems redundant, given you already seach for `is_encryption_key` above

Reply:
Iteration above is over subkeys, this is a check for *primary* key.

Reply:
I have renamed `k` into `subkey` and added a comment to make it more clear.